File: server/groqService.js
import Groq from 'groq-sdk';

export class GroqAIService {
  constructor(apiKey) {
    this.groq = new Groq({ apiKey });
    this.conversationHistory = [];
  }

  async chat(userMessage, systemPrompt = null) {
    const messages = [];

    if (systemPrompt) {
      messages.push({
        role: 'system',
        content: systemPrompt
      });
    }

    // Add conversation history
    messages.push(...this.conversationHistory);

    // Add new user message
    messages.push({
      role: 'user',
      content: userMessage
    });

    try {
      const completion = await this.groq.chat.completions.create({
        messages,
        model: 'llama-3.3-70b-versatile', // Using available Groq model
        temperature: 0.7,
        max_tokens: 8000,
      });

      const assistantMessage = completion.choices[0].message.content;

      // Update conversation history
      this.conversationHistory.push({
        role: 'user',
        content: userMessage
      });
      this.conversationHistory.push({
        role: 'assistant',
        content: assistantMessage
      });

      // Keep history manageable (last 10 messages)
      if (this.conversationHistory.length > 20) {
        this.conversationHistory = this.conversationHistory.slice(-20);
      }

      return assistantMessage;
    } catch (error) {
      console.error('Groq API error:', error);
      throw new Error('Failed to get AI response: ' + error.message);
    }
  }

  clearHistory() {
    this.conversationHistory = [];
  }
}

File: client/groq_service.py
from groq import Groq
import logging

logger = logging.getLogger(__name__)

class GroqAIService:
    def __init__(self, api_key):
        self.groq = Groq(api_key=api_key)
        self.conversation_history = []

    def chat(self, user_message, system_prompt=None):
        """Send a chat message to Groq API and get response"""
        messages = []

        if system_prompt:
            messages.append({
                'role': 'system',
                'content': system_prompt
            })

        # Add conversation history
        messages.extend(self.conversation_history)

        # Add new user message
        messages.append({
            'role': 'user',
            'content': user_message
        })

        try:
            completion = self.groq.chat.completions.create(
                messages=messages,
                model='llama-3.3-70b-versatile',
                temperature=0.7,
                max_tokens=8000,
            )

            assistant_message = completion.choices[0].message.content

            # Update conversation history
            self.conversation_history.append({
                'role': 'user',
                'content': user_message
            })
            self.conversation_history.append({
                'role': 'assistant',
                'content': assistant_message
            })

            # Keep history manageable (last 20 messages)
            if len(self.conversation_history) > 20:
                self.conversation_history = self.conversation_history[-20:]

            return assistant_message
        except Exception as e:
            logger.error(f'Groq API error: {e}')
            raise Exception(f'Failed to get AI response: {str(e)}')

    def clear_history(self):
        """Clear conversation history"""
        self.conversation_history = []
